smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -303,11 +304,13 @@ private void writeHeaders(
             for (HttpBinding binding : bindingIndex.getRequestBindings(operation, Location.PREFIX_HEADERS)) {
                 String memberName = symbolProvider.toMemberName(binding.getMember());
                 writer.openBlock("if (input.$L !== undefined) {", "}", memberName, () -> {
-                    Shape target = index.getShape(binding.getMember().getTarget()).get();
+                    MapShape prefixMap = index.getShape(binding.getMember().getTarget()).get().asMapShape().get();
```

Comment:
Lots of this can use `expectShape` rather than calling get on an optional. It should also be migrated to making calls directly on Model rather than going through ShapeIndex.